Add ability to specify LUT file paths via command argument

CrendKing opened the pull request:
DwmLut has some useful command arguments right now. However, if user wants to switch between different lut files via command, he has to edit config.xml then run with `-apply`.

This is attempt to fill the void by supporting two extra arguments "-sdr" and "-hdr". Example:
```
DwmLutGUI.exe -sdr "1:C:\luts\1\sdr.lut;2:C:\luts\2\sdr.lut" -hdr "1:C:\luts\1\hdr.lut;2:C:\luts\2\hdr.lut" -apply -exit
```
This example sets monitor 1's SDR lut to "C:\luts\1\sdr.lut", and monitor 2's SDR lut to "C:\luts\2\sdr.lut". Similar to HDR luts.

Also, paths set via this way does not permanently change the content of config.xml. This suits the "temporary" nature of this argument.

What do you think?

ledoge commented:
I like the idea and the syntax! Though I'm not sure how the interaction with the config.xml file should work. Right now it seems like the config file is still taken as the base, then the command line arguments override certain LUTs but without triggering a save, and any further change in the UI would then include the LUTs set via the command line in the config. Maybe there should be more arguments for whether the config should be used as the base or not, and whether further changes should result in the config file being saved? On the other hand that might be overcomplicating things too much.

At least for now, to avoid some of these issues, maybe it would make sense to require the `-apply` and `-exit` flags?

Also, I haven't been able to test your code yet, but from looking at, I believe that clearing the LUT/setting "nothing" as the LUT (e.g. `-sdr "1:"`) would result in the LUT path being set to `""` instead of `null`, which is what it should be.

CrendKing commented:
> overcomplicating things too much

I think so. I meant this to be a simple change.

> `-apply` and `-exit` flags

Done.

> set to `""` instead of `null`

Fixed.

CrendKing commented:
Added handling of some corner cases.

Also changed to use explicit types. If not desired, let me know and I'll revert. I personally like to always use explicit types for better readability.

I'll squash all commits when the review is done.

ledoge commented:
I think this might not be such a good idea after all, as it can't be used with setups where the number of monitors is inconsistent (or the number is the same, but the actual monitors are different). Ideally the command line arguments would include unique identifiers instead of just the index of the monitor in the list, but the problem with that is that I just had to switch to using the device path to uniquely identify monitors, as the UID wasn't enough... Those device paths are probably way too long to use in command line stuff like this though. Not sure what the solution is.

CrendKing commented:
I agree with your concern. I think the id of `DISPLAYCONFIG_PATH_TARGET_INFO` could change, so eventually you might want to have the monitor's name in the config.xml. Only when that's done, it makes sense to think about command line syntax. Is it gonna be too long? Could be if user has like 3 monitors. May be not so much for just one monitor.

Feel free to close this. I still think some sort of command line is nice.
